src/LibraryViewExtensionMSWebBrowser/LibraryViewCustomization.cs

```diff
@@ -0,0 +1,264 @@
+        public Stream ToJSONStream()
+        {
+            return root.ToJSONStream();
+        }
+
+        /// <summary>
+        /// Serializes the current specification to JSON stream and optionally replaces icon urls with images as base64 data
+        /// </summary>
+        public Stream ToJSONStream(bool replaceIconURLWithData, IconResourceProvider iconResourceProvider = null)
+        {
+            return ToJSONStream(root, replaceIconURLWithData, iconResourceProvider);
+        }
+
+
+        /// <summary>
+        /// Notifies when a resource stream is registered. This event is
+        /// used by ResourceHandlerFactory to register the resource handler
+        /// for provided url path.
+        /// </summary>
+        public event Action<string, Stream> ResourceStreamRegistered;
+
+        /// <summary>
+        /// Registers a given resource stream for a given url. If registered 
+        /// successfully the requested resource is returned from the given stream.
+        /// </summary>
+        /// <param name="urlpath">relative path of url including 
+        /// extension of the resource, e.g. /myicons/xicon.png</param>
+        /// <param name="resource">resource data stream</param>
+        /// <returns>True if the operation was successful</returns>
+        public bool RegisterResourceStream(string urlpath, Stream resource)
+        {
+            var extension = Path.GetExtension(urlpath);
+            if (string.IsNullOrEmpty(extension)) return false;
+
+            resources.Add(urlpath, resource);
+
+            var handler = ResourceStreamRegistered;
+            if (handler != null)
+            {
+                handler(urlpath, resource);
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// To be called by host application to request CEF shutdown
+        /// </summary>
+        public void OnAppShutdown()
+        {
+
+        }
+
+        /// <summary>
+        /// IDisposable implementation
+        /// </summary>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        private void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                foreach (var item in resources)
+                {
+                    item.Value.Dispose();
+                }
+                resources.Clear();
+            }
+        }
+
+        //layoutSpec helpers
+        /// <summary>
+        /// Serializes the layout specification to json stream and
+        /// optionally replaces icon urls with image data as base64 encoded strings
+        /// </summary>
+        /// <returns></returns>
+        private Stream ToJSONStream(LayoutSpecification spec, bool replaceIconURLWithData, IconResourceProvider iconProvider = null)
+        {
+            var ms = new MemoryStream();
+            var sw = new StreamWriter(ms);
+            if (replaceIconURLWithData && iconProvider != null)
```

Comment:
option of replacing images in the layoutspec - these would be for categories  - currently this path is not used.